class-gf-razorpay.php

```diff
@@ -587,15 +570,116 @@ public function billing_info_fields()
 
     public function init()
     {
+        wp_register_script('rzp', plugin_dir_url(__FILE__)  . 'assets/js/custom.js', null, null);
+        wp_enqueue_script('rzp');
+
         add_filter( 'gform_notification_events', array( $this, 'notification_events' ), 10, 2 );
 
+        add_action( 'wp_ajax_get_data', 'get_data' );
+        add_action( 'wp_ajax_nopriv_get_data', 'get_data' );
         // Supports frontend feeds.
         $this->_supports_frontend_feeds = true;
 
         parent::init();
 
     }
 
+    public function auto_enable_webhook()
+    { 
+            $webhookExist = false;
+            $webhookUrl = esc_url(admin_url('admin-post.php')) . '?action=gf_razorpay_webhook';
+            $enabled = true;
+            $alphanumericString = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ-=~!@#$%^&*()_+,./<>?;:[]{}|abcdefghijklmnopqrstuvwxyz';
+            $secret = substr(str_shuffle($alphanumericString), 0, 20);
+            update_option('rzp_webhook_secret', $secret);
+            $getWebhookFlag =  get_option('gf_webhook_enable_flag');
+            $time = time();
+
+            if (empty($getWebhookFlag))
+            {
+                add_option('gf_webhook_enable_flag', $time);
+            }
+            else
+            {
+                update_option('gf_webhook_enable_flag', $time);
+            }
+          
+            $skip = 0;
+            $count = 10;
+            $webhookItems= [];
+
+            do {
+                $webhook = $this->webhookAPI("GET", "webhooks?count=".$count."&skip=".$skip);
+                $skip += 10;
+                if ($webhook['count'] > 0)
+                {
+                    foreach ($webhook['items'] as $key => $value)
+                    {  
+                        if ($value['url'] === $webhookUrl)
+                        { 
+                            foreach ($value['events'] as $evntkey => $evntval)
+                            {
+                                if (($evntval == 1) and  
+                                    (in_array($evntkey, $this->supportedWebhookEvents) === true))
+                                {
+                                    $this->defaultWebhookEvents[$evntkey] =  true;
+                                }
+                            }
+                            $webhookExist  = true;
+                            $webhookId     = $value['id'];
```

Comment:
@jampulanaveen loop continues after webhook url is found as there is no break statement